RealmSwift/Tests/ObjectTests.swift

```diff
@@ -756,6 +1105,62 @@ class ObjectTests: TestCase {
         token.invalidate()
     }
 
+    func testOptionalPropertyKeyPathNotifications() {
```

Comment:
There are different tests in one here as well.

Reply:
I've reorganized the other tests. But I'm inclined to leave this one:
1) The other tests' expectations are more distinct in what code path/behavior is being tested.
2) The expectations for this test are dependent on each other (going from nil -> not nil -> back to nil), so there'd be more repeating/work.
3) Stays in precedent with the pre-key path test. (Of course we could always change those, but 1 and 2 applies to them too).